### PR TITLE
#89 Improve sorting of apps

### PR DESCRIPTION
**Pull request by jo-elimuai**
- Sort apps in the same order as received from the server

### CODE REVIEW

File: app/src/main/java/ai/elimu/appstore/dao/CustomDaoMaster.java [hunk 1]
```diff
@@ -41,6 +42,11 @@ public void onUpgrade(Database db, int oldVersion, int newVersion) {
                 // Add checksumMd5
                 DbMigrationHelper.migrate(db, ApplicationVersionDao.class);
             }
+
+            if (oldVersion < 2001000) {
+                // Add listOrder
```
Comment (tuancoltech) [inline app/src/main/java/ai/elimu/appstore/dao/CustomDaoMaster.java]: We are doing the same thing for 3 if blocks.
What about merging into 1, and leave some comments about data migration history of each?

Comment (tuancoltech) [reply app/src/main/java/ai/elimu/appstore/dao/CustomDaoMaster.java]: **DbMigrationHelper.migrate** will even be executed 3 times if oldVersion < 2000003.

Comment (jo-elimuai) [reply app/src/main/java/ai/elimu/appstore/dao/CustomDaoMaster.java]: We are only migrating the `Application` table once. Also, in this case, readability for future developers is more important than code efficiency. We need to get the upgrades from previous versions right without the app crashing.

Comment (jo-elimuai) [reply app/src/main/java/ai/elimu/appstore/dao/CustomDaoMaster.java]: If the user has not updated the app, and multiple releases of the app have been made in the meantime, it's absolutely fine if the `migrate` method is execute more than once. In fact, it should run the upgrade from one version to the next, without jumping to the newest version all at once.

**Pull request merged**